upstream.go

```diff
@@ -204,9 +204,30 @@ func (uc *upstreamConn) handleMessage(msg *irc.Message) error {
 	case "NOTICE":
 		uc.logger.Print(msg)
 
-		uc.forEachDownstream(func(dc *downstreamConn) {
-			dc.SendMessage(msg)
-		})
+		if msg.Prefix.User == "" && msg.Prefix.Host == "" { // server message
+			uc.forEachDownstream(func(dc *downstreamConn) {
+				dc.SendMessage(msg)
+			})
+		} else { // regular user NOTICE
+			var nick, text string
+			if err := parseMessageParams(msg, &nick, &text); err != nil {
+				return err
+			}
+
+			target := nick
+			if nick == uc.nick {
```

Comment:
Why do we have this special case? If the target is ourselves, why should we change the target to the source of the NOTICE message?

Reply:
Oh, this is only used for logs. Can we move this whole code block in the commit adding support for logs? (`git rebase -i`, edit both commits, cut the block from this commit, `git rebase --continue`, paste the block to the next one)

Reply:
Ugh, GitHub shows these lines as added in both the NOTICE commit and the logging commit. Not sure why.

Reply:
Should be fixed now.